dhis-2/dhis-services/dhis-service-acl/src/main/java/org/hisp/dhis/security/acl/DefaultAclService.java

```diff
@@ -413,17 +482,23 @@ public Class<? extends IdentifiableObject> classForType(String type) {
   }
 
   @Override
-  @SuppressWarnings("unchecked")
   public <T extends IdentifiableObject> Access getAccess(T object, User user) {
+    return getAccess(object, UserDetails.fromUser(user));
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <T extends IdentifiableObject> Access getAccess(T object, UserDetails userDetails) {
     return object == null
         ? new Access(true)
-        : getAccess(object, user, HibernateProxyUtils.getRealClass(object));
+        : getAccess(object, userDetails, HibernateProxyUtils.getRealClass(object));
   }
 
   @Override
   public <T extends IdentifiableObject> Access getAccess(
-      T object, User user, Class<? extends T> objType) {
-    if (user == null || user.isSuper()) {
+      T object, UserDetails userDetails, Class<? extends T> objType) {
+
+    if (userDetails == null || isSuper(userDetails)) {
```

Comment:
is this similar to the old pattern we're trying to remove? (treating null user as super, now treating null userDetails as super?)